photon-ml/src/main/scala/com/linkedin/photon/ml/model/Coefficients.scala

```diff
@@ -50,13 +50,18 @@ protected[ml] case class Coefficients(means: Vector[Double], variancesOption: Op
     stringBuilder.toString()
   }
 
+  // TODO: what should we do about the epsilon used in this comparison?
   override def equals(that: Any): Boolean = {
+
+    def cmp(a1: Vector[Double], a2: Vector[Double]): Boolean =
+      (a1.toArray zip a2.toArray).foldLeft(true) { case (v, (m1, m2)) => v & math.abs(m2 - m1) < 1e-6}
```

Comment:
More concise, and better to use the tolerance constant than a magic number:

```
(a1.toArray zip a2.toArray).forall {
  case ((m1, m2)) => math.abs(m2 - m1) < MathConst.HIGH_PRECISION_TOLERANCE_THRESHOLD
}
```

Also, we should give this a more descriptive name and move it to VectorUtils.